src/services/data_model_engine.js

```diff
@@ -226,13 +226,28 @@ export default class DataModelEngine {
 
   async downloadBundle(bundleId, sheltererId) {
     const nodeUrl = await this.rolesRepository.nodeUrl(sheltererId);
+
+    const bundleMetadata = await this.bundleDownloader.downloadBundleMetadata(nodeUrl, bundleId);
+    if (!bundleMetadata) {
+      throw new Error('Could not fetch the bundle metadata from the shelterer');
+    }
+
     const bundle = await this.bundleDownloader.downloadBundle(nodeUrl, bundleId);
     if (!bundle) {
       throw new Error('Could not fetch the bundle from the shelterer');
     }
+
     this.bundleBuilder.validateBundle(bundle);
     await this.uploadRepository.verifyBundle(bundle);
```

Comment:
IMO: We should also validate the bundle metadata. Could you add a task for this in Asana?